### ISSUE TITLE
EDSC-1070: Integrate AST_L1T test granule imagery into Earthdata Search

### ISSUE THREAD

**Issue by earthdata-tasktop** [nasa/edsc-search-test-config-3#222]
This should be just a matter of dusting off existing code to show GIBS granule imagery.

**Comment by earthdata-tasktop** [comment]
(Comment from Francell Gray):

Executes the UAT granule search:  

[https://search.uat.earthdata.nasa.gov/search/granules?cmr_env=ops&amp;amp;p=C1000000320-LPDAAC_ECS&amp;amp;m=30.276123046875!-104.5810546875!7!1!0!0%2C2&amp;amp;tl=1455697014!2!1455728400!1455731999&amp;amp;q=ast](https://search.uat.earthdata.nasa.gov/search/granules?cmr_env=ops&p=C1000000320-LPDAAC_ECS&m=30.276123046875!-104.5810546875!7!1!0!0%2C2&tl=1455697014!2!1455728400!1455731999&q=ast)

Verified that imagery was displayed for 10 of the 15 granules returned.  See screen shot.

**Comment by earthdata-tasktop** [comment]
(Comment from Abdul Khan):

This looks really good.  I am wondering if it would be possible to re-project the imagery so it fits in the bounding box of the granule.  I also wonder how GIBS generated these png files.  These algorithms could fit very well into ESI so that users could order a large number of GIBS imagery.

**Comment by earthdata-tasktop** [comment]
(Comment from Patrick Quinn):

To test, you need to look at AST_L1T for the relatively small number of granules enabled for this testing. Only 10 granules have imagery. They are consecutive granules on February 17th, 2016 during the hour of 17:00 UTC.

This filter will produce the correct search and map view:  

[https://search.uat.earthdata.nasa.gov/search/granules?cmr_env=ops&amp;amp;p=C1000000320-LPDAAC_ECS&amp;amp;m=30.276123046875!-104.5810546875!7!1!0!0%2C2&amp;amp;tl=1455697014!2!1455728400!1455731999&amp;amp;q=ast_l1t](https://search.uat.earthdata.nasa.gov/search/granules?cmr_env=ops&p=C1000000320-LPDAAC_ECS&m=30.276123046875!-104.5810546875!7!1!0!0%2C2&tl=1455697014!2!1455728400!1455731999&q=ast_l1t)

You should see granule imagery for 10 of the 15 granules in that time window. The clipping behavior proves that it's per-granule imagery.

**Comment by earthdata-tasktop** [comment]
(Comment from Patrick Quinn):

Info from Matthew Cechini:

Patrick,  

   At long last, we have the AST_L1T granules in UAT.  Sample access URLs are here, depending on which endpoint you want to access them through.  Slack me if you want to chat about what the differences in these endpoints are.

[http://uat.gibs.earthdata.nasa.gov/wmts/epsg4326/std/ASTER_L1T_Radiance_Terrain_Corrected_v3_STD/default/2016-02-17T17:45:56Z/31.25m/7/27/33.png](http://uat.gibs.earthdata.nasa.gov/wmts/epsg4326/std/ASTER_L1T_Radiance_Terrain_Corrected_v3_STD/default/2016-02-17T17:45:56Z/31.25m/7/27/33.png)  

[http://uat.gibs.earthdata.nasa.gov/wmts/epsg4326/best/ASTER_L1T_Radiance_Terrain_Corrected/default/2016-02-17T17:45:56Z/31.25m/7/27/33.png](http://uat.gibs.earthdata.nasa.gov/wmts/epsg4326/best/ASTER_L1T_Radiance_Terrain_Corrected/default/2016-02-17T17:45:56Z/31.25m/7/27/33.png)  

[http://uat.gibs.earthdata.nasa.gov/wmts/epsg4326/all/ASTER_L1T_Radiance_Terrain_Corrected_v3_STD/default/2016-02-17T17:45:56Z/31.25m/7/27/33.png](http://uat.gibs.earthdata.nasa.gov/wmts/epsg4326/all/ASTER_L1T_Radiance_Terrain_Corrected_v3_STD/default/2016-02-17T17:45:56Z/31.25m/7/27/33.png)  

[http://uat.gibs.earthdata.nasa.gov/wmts/epsg4326/all/ASTER_L1T_Radiance_Terrain_Corrected/default/2016-02-17T17:45:56Z/31.25m/7/27/33.png](http://uat.gibs.earthdata.nasa.gov/wmts/epsg4326/all/ASTER_L1T_Radiance_Terrain_Corrected/default/2016-02-17T17:45:56Z/31.25m/7/27/33.png)

The following date/time values will give you valid granule imagery.  We are not snapping to the nearest granule if you provide a non-precise time.  I also don’t have the Granule Ids handy, but a quick CMR query will yield those.

2016-02-17T17:44:27Z  

2016-02-17T17:44:54Z  

2016-02-17T17:44:45Z  

2016-02-17T17:44:36Z  

2016-02-17T17:45:03Z  

2016-02-17T17:45:12Z  

2016-02-17T17:45:20Z  

2016-02-17T17:45:29Z  

2016-02-17T17:45:38Z  

2016-02-17T17:45:47Z  

2016-02-17T17:45:56Z

Let me know if you have any questions.  It would be nice to have something by ESIP (mid-July) to demo, if that fits into your schedule.

Thanks,  

Matt

--

the std/nrt GIBS endpoints provide direct access to an imagery product/version.  The best endpoint provides the “Best” image (nrt vs std) depending on what we have for a date.  Best really isn’t applicable for AST_L1T, since we only have STD.  The all/ endpoint is a superset of the std/nrt/best endpoints.  map1.vis redirects to best/ now.

**Comment by earthdata-tasktop** [comment]
(Comment from Patrick Quinn):

Approved at sprint 1.20 planning

**Comment by earthdata-tasktop** [comment]
(Comment from Patrick Quinn):

Waiting on GIBS to be deployed to its UAT. ETA next week.

**Comment by earthdata-tasktop** [comment]
(Comment from Patrick Quinn):

Info from Matthew Cechini:

Patrick,  

   At long last, we have the AST_L1T granules in UAT.  Sample access URLs are here, depending on which endpoint you want to access them through.  Slack me if you want to chat about what the differences in these endpoints are.

[http://uat.gibs.earthdata.nasa.gov/wmts/epsg4326/std/ASTER_L1T_Radiance_Terrain_Corrected_v3_STD/default/2016-02-17T17:45:56Z/31.25m/7/27/33.png](http://uat.gibs.earthdata.nasa.gov/wmts/epsg4326/std/ASTER_L1T_Radiance_Terrain_Corrected_v3_STD/default/2016-02-17T17:45:56Z/31.25m/7/27/33.png)  

[http://uat.gibs.earthdata.nasa.gov/wmts/epsg4326/best/ASTER_L1T_Radiance_Terrain_Corrected/default/2016-02-17T17:45:56Z/31.25m/7/27/33.png](http://uat.gibs.earthdata.nasa.gov/wmts/epsg4326/best/ASTER_L1T_Radiance_Terrain_Corrected/default/2016-02-17T17:45:56Z/31.25m/7/27/33.png)  

[http://uat.gibs.earthdata.nasa.gov/wmts/epsg4326/all/ASTER_L1T_Radiance_Terrain_Corrected_v3_STD/default/2016-02-17T17:45:56Z/31.25m/7/27/33.png](http://uat.gibs.earthdata.nasa.gov/wmts/epsg4326/all/ASTER_L1T_Radiance_Terrain_Corrected_v3_STD/default/2016-02-17T17:45:56Z/31.25m/7/27/33.png)  

[http://uat.gibs.earthdata.nasa.gov/wmts/epsg4326/all/ASTER_L1T_Radiance_Terrain_Corrected/default/2016-02-17T17:45:56Z/31.25m/7/27/33.png](http://uat.gibs.earthdata.nasa.gov/wmts/epsg4326/all/ASTER_L1T_Radiance_Terrain_Corrected/default/2016-02-17T17:45:56Z/31.25m/7/27/33.png)

The following date/time values will give you valid granule imagery.  We are not snapping to the nearest granule if you provide a non-precise time.  I also don’t have the Granule Ids handy, but a quick CMR query will yield those.

2016-02-17T17:44:27Z  

2016-02-17T17:44:54Z  

2016-02-17T17:44:45Z  

2016-02-17T17:44:36Z  

2016-02-17T17:45:03Z  

2016-02-17T17:45:12Z  

2016-02-17T17:45:20Z  

2016-02-17T17:45:29Z  

2016-02-17T17:45:38Z  

2016-02-17T17:45:47Z  

2016-02-17T17:45:56Z

Let me know if you have any questions.  It would be nice to have something by ESIP (mid-July) to demo, if that fits into your schedule.

Thanks,  

Matt

--

the std/nrt GIBS endpoints provide direct access to an imagery product/version.  The best endpoint provides the “Best” image (nrt vs std) depending on what we have for a date.  Best really isn’t applicable for AST_L1T, since we only have STD.  The all/ endpoint is a superset of the std/nrt/best endpoints.  map1.vis redirects to best/ now.

**Comment by earthdata-tasktop** [comment]
(Comment from Patrick Quinn):

Approved at sprint 1.20 planning

**Comment by earthdata-tasktop** [comment]
(Comment from Patrick Quinn):

Waiting on GIBS to be deployed to its UAT. ETA next week.

**Comment by earthdata-tasktop** [comment]
(Comment from Patrick Quinn):

Approved at sprint 1.20 planning

**Comment by earthdata-tasktop** [comment]
(Comment from Patrick Quinn):

Waiting on GIBS to be deployed to its UAT. ETA next week.

**Comment by earthdata-tasktop** [comment]
(Comment from Patrick Quinn):

Waiting on GIBS to be deployed to its UAT. ETA next week.

**Comment by earthdata-tasktop** [comment]
(Comment from Abdul Khan):

This looks really good.  I am wondering if it would be possible to re-project the imagery so it fits in the bounding box of the granule.  I also wonder how GIBS generated these png files.  These algorithms could fit very well into ESI so that users could order a large number of GIBS imagery.

**Comment by earthdata-tasktop** [comment]
(Comment from Patrick Quinn):

To test, you need to look at AST_L1T for the relatively small number of granules enabled for this testing. Only 10 granules have imagery. They are consecutive granules on February 17th, 2016 during the hour of 17:00 UTC.

This filter will produce the correct search and map view:  

[https://search.uat.earthdata.nasa.gov/search/granules?cmr_env=ops&amp;amp;p=C1000000320-LPDAAC_ECS&amp;amp;m=30.276123046875!-104.5810546875!7!1!0!0%2C2&amp;amp;tl=1455697014!2!1455728400!1455731999&amp;amp;q=ast_l1t](https://search.uat.earthdata.nasa.gov/search/granules?cmr_env=ops&p=C1000000320-LPDAAC_ECS&m=30.276123046875!-104.5810546875!7!1!0!0%2C2&tl=1455697014!2!1455728400!1455731999&q=ast_l1t)

You should see granule imagery for 10 of the 15 granules in that time window. The clipping behavior proves that it's per-granule imagery.

**Comment by earthdata-tasktop** [comment]
(Comment from Patrick Quinn):

Info from Matthew Cechini:

Patrick,  

   At long last, we have the AST_L1T granules in UAT.  Sample access URLs are here, depending on which endpoint you want to access them through.  Slack me if you want to chat about what the differences in these endpoints are.

[http://uat.gibs.earthdata.nasa.gov/wmts/epsg4326/std/ASTER_L1T_Radiance_Terrain_Corrected_v3_STD/default/2016-02-17T17:45:56Z/31.25m/7/27/33.png](http://uat.gibs.earthdata.nasa.gov/wmts/epsg4326/std/ASTER_L1T_Radiance_Terrain_Corrected_v3_STD/default/2016-02-17T17:45:56Z/31.25m/7/27/33.png)  

[http://uat.gibs.earthdata.nasa.gov/wmts/epsg4326/best/ASTER_L1T_Radiance_Terrain_Corrected/default/2016-02-17T17:45:56Z/31.25m/7/27/33.png](http://uat.gibs.earthdata.nasa.gov/wmts/epsg4326/best/ASTER_L1T_Radiance_Terrain_Corrected/default/2016-02-17T17:45:56Z/31.25m/7/27/33.png)  

[http://uat.gibs.earthdata.nasa.gov/wmts/epsg4326/all/ASTER_L1T_Radiance_Terrain_Corrected_v3_STD/default/2016-02-17T17:45:56Z/31.25m/7/27/33.png](http://uat.gibs.earthdata.nasa.gov/wmts/epsg4326/all/ASTER_L1T_Radiance_Terrain_Corrected_v3_STD/default/2016-02-17T17:45:56Z/31.25m/7/27/33.png)  

[http://uat.gibs.earthdata.nasa.gov/wmts/epsg4326/all/ASTER_L1T_Radiance_Terrain_Corrected/default/2016-02-17T17:45:56Z/31.25m/7/27/33.png](http://uat.gibs.earthdata.nasa.gov/wmts/epsg4326/all/ASTER_L1T_Radiance_Terrain_Corrected/default/2016-02-17T17:45:56Z/31.25m/7/27/33.png)

The following date/time values will give you valid granule imagery.  We are not snapping to the nearest granule if you provide a non-precise time.  I also don’t have the Granule Ids handy, but a quick CMR query will yield those.

2016-02-17T17:44:27Z  

2016-02-17T17:44:54Z  

2016-02-17T17:44:45Z  

2016-02-17T17:44:36Z  

2016-02-17T17:45:03Z  

2016-02-17T17:45:12Z  

2016-02-17T17:45:20Z  

2016-02-17T17:45:29Z  

2016-02-17T17:45:38Z  

2016-02-17T17:45:47Z  

2016-02-17T17:45:56Z

Let me know if you have any questions.  It would be nice to have something by ESIP (mid-July) to demo, if that fits into your schedule.

Thanks,  

Matt

--

the std/nrt GIBS endpoints provide direct access to an imagery product/version.  The best endpoint provides the “Best” image (nrt vs std) depending on what we have for a date.  Best really isn’t applicable for AST_L1T, since we only have STD.  The all/ endpoint is a superset of the std/nrt/best endpoints.  map1.vis redirects to best/ now.

**Comment by earthdata-tasktop** [comment]
(Comment from Patrick Quinn):

Info from Matthew Cechini:

Patrick,  

   At long last, we have the AST_L1T granules in UAT.  Sample access URLs are here, depending on which endpoint you want to access them through.  Slack me if you want to chat about what the differences in these endpoints are.

[http://uat.gibs.earthdata.nasa.gov/wmts/epsg4326/std/ASTER_L1T_Radiance_Terrain_Corrected_v3_STD/default/2016-02-17T17:45:56Z/31.25m/7/27/33.png](http://uat.gibs.earthdata.nasa.gov/wmts/epsg4326/std/ASTER_L1T_Radiance_Terrain_Corrected_v3_STD/default/2016-02-17T17:45:56Z/31.25m/7/27/33.png)  

[http://uat.gibs.earthdata.nasa.gov/wmts/epsg4326/best/ASTER_L1T_Radiance_Terrain_Corrected/default/2016-02-17T17:45:56Z/31.25m/7/27/33.png](http://uat.gibs.earthdata.nasa.gov/wmts/epsg4326/best/ASTER_L1T_Radiance_Terrain_Corrected/default/2016-02-17T17:45:56Z/31.25m/7/27/33.png)  

[http://uat.gibs.earthdata.nasa.gov/wmts/epsg4326/all/ASTER_L1T_Radiance_Terrain_Corrected_v3_STD/default/2016-02-17T17:45:56Z/31.25m/7/27/33.png](http://uat.gibs.earthdata.nasa.gov/wmts/epsg4326/all/ASTER_L1T_Radiance_Terrain_Corrected_v3_STD/default/2016-02-17T17:45:56Z/31.25m/7/27/33.png)  

[http://uat.gibs.earthdata.nasa.gov/wmts/epsg4326/all/ASTER_L1T_Radiance_Terrain_Corrected/default/2016-02-17T17:45:56Z/31.25m/7/27/33.png](http://uat.gibs.earthdata.nasa.gov/wmts/epsg4326/all/ASTER_L1T_Radiance_Terrain_Corrected/default/2016-02-17T17:45:56Z/31.25m/7/27/33.png)

The following date/time values will give you valid granule imagery.  We are not snapping to the nearest granule if you provide a non-precise time.  I also don’t have the Granule Ids handy, but a quick CMR query will yield those.

2016-02-17T17:44:27Z  

2016-02-17T17:44:54Z  

2016-02-17T17:44:45Z  

2016-02-17T17:44:36Z  

2016-02-17T17:45:03Z  

2016-02-17T17:45:12Z  

2016-02-17T17:45:20Z  

2016-02-17T17:45:29Z  

2016-02-17T17:45:38Z  

2016-02-17T17:45:47Z  

2016-02-17T17:45:56Z

Let me know if you have any questions.  It would be nice to have something by ESIP (mid-July) to demo, if that fits into your schedule.

Thanks,  

Matt

--

the std/nrt GIBS endpoints provide direct access to an imagery product/version.  The best endpoint provides the “Best” image (nrt vs std) depending on what we have for a date.  Best really isn’t applicable for AST_L1T, since we only have STD.  The all/ endpoint is a superset of the std/nrt/best endpoints.  map1.vis redirects to best/ now.

**Comment by earthdata-tasktop** [comment]
(Comment from Patrick Quinn):

Approved at sprint 1.20 planning

**Comment by earthdata-tasktop** [comment]
(Comment from Patrick Quinn):

Waiting on GIBS to be deployed to its UAT. ETA next week.

**Comment by earthdata-tasktop** [comment]
(Comment from Patrick Quinn):

Approved at sprint 1.20 planning

**Comment by earthdata-tasktop** [comment]
(Comment from Patrick Quinn):

Waiting on GIBS to be deployed to its UAT. ETA next week.

**Comment by earthdata-tasktop** [comment]
(Comment from Patrick Quinn):

To test, you need to look at AST_L1T for the relatively small number of granules enabled for this testing. Only 10 granules have imagery. They are consecutive granules on February 17th, 2016 during the hour of 17:00 UTC.

This filter will produce the correct search and map view:  

[https://search.uat.earthdata.nasa.gov/search/granules?cmr_env=ops&amp;amp;p=C1000000320-LPDAAC_ECS&amp;amp;m=30.276123046875!-104.5810546875!7!1!0!0%2C2&amp;amp;tl=1455697014!2!1455728400!1455731999&amp;amp;q=ast_l1t](https://search.uat.earthdata.nasa.gov/search/granules?cmr_env=ops&p=C1000000320-LPDAAC_ECS&m=30.276123046875!-104.5810546875!7!1!0!0%2C2&tl=1455697014!2!1455728400!1455731999&q=ast_l1t)

You should see granule imagery for 10 of the 15 granules in that time window. The clipping behavior proves that it's per-granule imagery.

**Comment by earthdata-tasktop** [comment]
(Comment from Patrick Quinn):

To test, you need to look at AST_L1T for the relatively small number of granules enabled for this testing. Only 10 granules have imagery. They are consecutive granules on February 17th, 2016 during the hour of 17:00 UTC.

This filter will produce the correct search and map view:  

[https://search.uat.earthdata.nasa.gov/search/granules?cmr_env=ops&amp;amp;p=C1000000320-LPDAAC_ECS&amp;amp;m=30.276123046875!-104.5810546875!7!1!0!0%2C2&amp;amp;tl=1455697014!2!1455728400!1455731999&amp;amp;q=ast_l1t](https://search.uat.earthdata.nasa.gov/search/granules?cmr_env=ops&p=C1000000320-LPDAAC_ECS&m=30.276123046875!-104.5810546875!7!1!0!0%2C2&tl=1455697014!2!1455728400!1455731999&q=ast_l1t)

You should see granule imagery for 10 of the 15 granules in that time window. The clipping behavior proves that it's per-granule imagery.

**Comment by earthdata-tasktop** [comment]
(Comment from Patrick Quinn):

Info from Matthew Cechini:

Patrick,  

   At long last, we have the AST_L1T granules in UAT.  Sample access URLs are here, depending on which endpoint you want to access them through.  Slack me if you want to chat about what the differences in these endpoints are.

[http://uat.gibs.earthdata.nasa.gov/wmts/epsg4326/std/ASTER_L1T_Radiance_Terrain_Corrected_v3_STD/default/2016-02-17T17:45:56Z/31.25m/7/27/33.png](http://uat.gibs.earthdata.nasa.gov/wmts/epsg4326/std/ASTER_L1T_Radiance_Terrain_Corrected_v3_STD/default/2016-02-17T17:45:56Z/31.25m/7/27/33.png)  

[http://uat.gibs.earthdata.nasa.gov/wmts/epsg4326/best/ASTER_L1T_Radiance_Terrain_Corrected/default/2016-02-17T17:45:56Z/31.25m/7/27/33.png](http://uat.gibs.earthdata.nasa.gov/wmts/epsg4326/best/ASTER_L1T_Radiance_Terrain_Corrected/default/2016-02-17T17:45:56Z/31.25m/7/27/33.png)  

[http://uat.gibs.earthdata.nasa.gov/wmts/epsg4326/all/ASTER_L1T_Radiance_Terrain_Corrected_v3_STD/default/2016-02-17T17:45:56Z/31.25m/7/27/33.png](http://uat.gibs.earthdata.nasa.gov/wmts/epsg4326/all/ASTER_L1T_Radiance_Terrain_Corrected_v3_STD/default/2016-02-17T17:45:56Z/31.25m/7/27/33.png)  

[http://uat.gibs.earthdata.nasa.gov/wmts/epsg4326/all/ASTER_L1T_Radiance_Terrain_Corrected/default/2016-02-17T17:45:56Z/31.25m/7/27/33.png](http://uat.gibs.earthdata.nasa.gov/wmts/epsg4326/all/ASTER_L1T_Radiance_Terrain_Corrected/default/2016-02-17T17:45:56Z/31.25m/7/27/33.png)

The following date/time values will give you valid granule imagery.  We are not snapping to the nearest granule if you provide a non-precise time.  I also don’t have the Granule Ids handy, but a quick CMR query will yield those.

2016-02-17T17:44:27Z  

2016-02-17T17:44:54Z  

2016-02-17T17:44:45Z  

2016-02-17T17:44:36Z  

2016-02-17T17:45:03Z  

2016-02-17T17:45:12Z  

2016-02-17T17:45:20Z  

2016-02-17T17:45:29Z  

2016-02-17T17:45:38Z  

2016-02-17T17:45:47Z  

2016-02-17T17:45:56Z

Let me know if you have any questions.  It would be nice to have something by ESIP (mid-July) to demo, if that fits into your schedule.

Thanks,  

Matt

--

the std/nrt GIBS endpoints provide direct access to an imagery product/version.  The best endpoint provides the “Best” image (nrt vs std) depending on what we have for a date.  Best really isn’t applicable for AST_L1T, since we only have STD.  The all/ endpoint is a superset of the std/nrt/best endpoints.  map1.vis redirects to best/ now.

**Comment by earthdata-tasktop** [comment]
(Comment from Patrick Quinn):

Approved at sprint 1.20 planning

**Comment by earthdata-tasktop** [comment]
(Comment from Patrick Quinn):

Waiting on GIBS to be deployed to its UAT. ETA next week.

**Comment by earthdata-tasktop** [comment]
(Comment from Patrick Quinn):

Waiting on GIBS to be deployed to its UAT. ETA next week.

**Comment by earthdata-tasktop** [comment]
(Comment from Patrick Quinn):

Approved at sprint 1.20 planning

**Comment by earthdata-tasktop** [comment]
(Comment from Patrick Quinn):

Waiting on GIBS to be deployed to its UAT. ETA next week.
